cmd/tnf/main.go

```diff
@@ -36,53 +38,97 @@ var (
 		Short: "adding new handler.",
 		RunE:  generateHandlerFiles,
 	}
+
+	defaultHandlersFolder = path.Join("pkg", "tnf", "handlers")
 )
 
-func generateHandlerFiles(cmd *cobra.Command, args []string) error {
-	handlername = args[0]
-	pathrelativetoroot = path.Join("..", "..")
-	handlerDirectory = path.Join(pathrelativetoroot, "pkg", "tnf", "handlers")
-	newHandlerDirectory := path.Join(handlerDirectory, handlername)
+func getHandlersDirectory() (string, error) {
+	handlersDirectory := os.Getenv(envHandlersFolder)
 
-	err := os.Mkdir(newHandlerDirectory, 0755)
-	if err != nil {
+	if handlersDirectory == "" {
+		log.Warnf("Environment variable %s not set. Handlers base folder will be set to ./%s",
+			envHandlersFolder, defaultHandlersFolder)
+
+		handlersDirectory = defaultHandlersFolder
+	} else {
+		log.Infof("Env var %s found. Handlers directory: %s", envHandlersFolder, handlersDirectory)
+	}
+
+	// Convert to absolute path.
+	if !path.IsAbs(handlersDirectory) {
+		cwd, err := os.Getwd()
+		if err != nil {
+			return "", err
+		}
+
+		handlersDirectory = path.Join(cwd, handlersDirectory)
+	}
+
+	return handlersDirectory, nil
+}
+
+func generateHandlerFilesFromTemplates(handlerTemplatesDirectory, newHandlerDirectory string, myhandler myHandler) error {
+	templateFilePath := path.Join(handlerTemplatesDirectory, "doc.tmpl")
+	renderedFileName := docFileName
+
+	if err := createfile(templateFilePath, renderedFileName, myhandler, newHandlerDirectory); err != nil {
 		return err
 	}
 
-	myhandler := myHandler{LowerHandlername: handlername, UpperHandlername: strings.Title(handlername)}
+	templateFilePath = path.Join(handlerTemplatesDirectory, "handler_test.tmpl")
+	renderedFileName = myhandler.LowerHandlername + "_test.go"
 
-	// pathfile this is the path of the file from template file that will creat
+	if err := createfile(templateFilePath, renderedFileName, myhandler, newHandlerDirectory); err != nil {
+		return err
+	}
 
-	pathfile := path.Join(handlerDirectory, "handler_template", "doc.tmpl")
-	namefile := "" + "doc.go"
-	err = createfile(pathfile, namefile, myhandler, newHandlerDirectory) // here creating file by doc.tmpl
-	if err != nil {
+	templateFilePath = path.Join(handlerTemplatesDirectory, "handler.tmpl")
+	renderedFileName = myhandler.LowerHandlername + ".go"
+
+	if err := createfile(templateFilePath, renderedFileName, myhandler, newHandlerDirectory); err != nil {
 		return err
```

Comment:
you probably can have a loop like this!
```
for s:= range []string{handler_test,handler.temp,doc.template}{
 // read string
 // replace handler with real_hander and temp with go
 // generate file
}
```

Reply:
Not sure if my new approach is exactly what you were suggesting, but at least it avoids the previous repetitive code.

Reply:
it's not the same, but it's good!